components/config/prefs.rs

```diff
@@ -56,11 +60,58 @@ pub fn pref_map() -> &'static Preferences<'static, Prefs> {
 }
 
 pub fn add_user_prefs(prefs: HashMap<String, PrefValue>) {
+    for (key, value) in prefs.iter() {
+        set_stylo_pref_ref(key, value);
+    }
     if let Err(error) = PREFS.set_all(prefs.into_iter()) {
         panic!("Error setting preference: {:?}", error);
     }
 }
 
+pub fn set_stylo_pref(key: &str, value: impl Into<PrefValue>) {
+    set_stylo_pref_ref(key, &value.into());
+}
+
+fn set_stylo_pref_ref(key: &str, value: &PrefValue) {
+    dbg!(key);
+    match value.try_into() {
+        Ok(StyloPrefValue::Bool(value)) => style_config::set_bool(key, value),
+        Ok(StyloPrefValue::Int(value)) => style_config::set_i32(key, value),
+        Err(TryFromPrefValueError::IntegerOverflow(value)) => {
+            warn!("Pref value too big for Stylo: {} ({})", key, value);
+        },
+        Err(TryFromPrefValueError::UnmappedType) => {},
```

Comment:
i was going to say this should be a debug message because there are many prefs that would hit this, then i found a bug (ffa68c0c585c875e112d6dabc5e462e350144669), then i found that logging doesn’t work this early anyway, the messages just vanish. i’ve added some comments to that effect in b968157cb525322f5357b670e435a537f2a220a1, and i’ll handle the todo in a separate patch.